spinalcordtoolbox/scripts/sct_label_utils.py

```diff
@@ -374,8 +375,19 @@ def launch_sagittal_viewer(img: Image, labels: Sequence[int], msg: str, previous
 
     params.subtitle = msg
 
+    # NB: `num_points` is used as the limit for how many points *should* be in the image.
+    # If this number is exceeded, `SagittalController` will throw a `TooManyPoints` warning.
```

Comment:
In my testing, I'm not seeing the associated message appearing in the GUI:
https://github.com/spinalcordtoolbox/spinalcordtoolbox/blob/bb89863a06036d6762af1d18f7899f128687a9c6/spinalcordtoolbox/gui/base.py#L391-L392

But I don't think Exceptions use this supposed `message` member? Indeed:
```py3
Python 3.8.15 (default, Nov 24 2022, 15:19:38) 
[GCC 11.2.0] :: Anaconda, Inc. on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> from spinalcordtoolbox.gui.base import TooManyPointsWarning
>>> str(TooManyPointsWarning())
''
>>> raise TooManyPointsWarning
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
spinalcordtoolbox.gui.base.TooManyPointsWarning
>>> raise TooManyPointsWarning("an actual message")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
spinalcordtoolbox.gui.base.TooManyPointsWarning: an actual message
```

Plus, I don't think it should subclass `StopIteration`, that's a very specific exception type used for Python's control flow handling.

In any case, the exception _does_ serve the purpose of preventing extra labels being created once the limit is reached.

Reply:
In contrast, I do see the following message if I create a label and try to undo too many times:
https://github.com/spinalcordtoolbox/spinalcordtoolbox/blob/bb89863a06036d6762af1d18f7899f128687a9c6/spinalcordtoolbox/gui/base.py#L364

(And if I try to undo _before_ creating any labels, the program just crashes, hmmm.)

Reply:
> (And if I try to undo before creating any labels, the program just crashes, hmmm.)

This is strange. Supposedly, the case you describe is handled by the `else` in [`.undo()`](https://github.com/spinalcordtoolbox/spinalcordtoolbox/blob/e7c82c359bfb0a24cc852fa4124701bd1b31d659/spinalcordtoolbox/gui/base.py#L355-L364):

The traceback here is:

```
Traceback (most recent call last):
  File "/home/joshua/repos/spinalcordtoolbox/spinalcordtoolbox/gui/sagittal.py", line 85, in on_undo
    self.labels.label = self._controller.label
AttributeError: 'SagittalController' object has no attribute 'label'
```

https://github.com/spinalcordtoolbox/spinalcordtoolbox/blob/e7c82c359bfb0a24cc852fa4124701bd1b31d659/spinalcordtoolbox/gui/sagittal.py#L77-L81

I can confirm that this happens for both `-create-viewer 1,1,1` and `-create-viewer 1,2,3`, so this seems to be a bug outside the scope of the changes in this PR. I'll create a new issue.